app/jobs/ingest_yaml_job.rb

```diff
@@ -115,4 +133,39 @@ def yaml_to_repo_map
     def thumbnail_path
       @thumbnail_path ||= @yaml[:thumbnail_path]
     end
+
+    # TESTING
+        def attach_files_to_work(work, file_sets)
+          acquire_lock_for(work.id) do
+            set_representative(work, file_sets.first) if file_sets.any?
+            set_thumbnail(work, file_sets.first) if file_sets.any?
+            # Ensure we have an up-to-date copy of the members association, so
+            # that we append to the end of the list.
+            work.reload unless work.new_record?
+            work.ordered_members = file_sets
```

Comment:
Changing = to << would make this more generic, for use in multiple file upload from the interface